File: mobile-app/src/screens/LockScreen.js
import { Alert, StyleSheet, Text, View } from "react-native";
import { Screen } from "../components/Screen";
import { PrimaryButton } from "../components/PrimaryButton";
import { useAuth } from "../auth/AuthProvider";
import { colors, spacing } from "../theme/tokens";

export function LockScreen() {
  const { unlock, signOut } = useAuth();

  async function handleUnlock() {
    const ok = await unlock();
    if (!ok) Alert.alert("Unlock failed", "Biometric authentication did not complete.");
  }

  return (
    <Screen scroll={false} contentStyle={styles.screen}>
      <View>
        <Text style={styles.title}>SchoolDom is locked</Text>
        <Text style={styles.copy}>Unlock with your device biometrics to continue.</Text>
      </View>
      <PrimaryButton title="Unlock" onPress={handleUnlock} />
      <PrimaryButton title="Use another account" tone="ghost" onPress={signOut} />
    </Screen>
  );
}

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    justifyContent: "center",
  },
  title: {
    color: colors.text,
    fontSize: 30,
    fontWeight: "900",
  },
  copy: {
    color: colors.muted,
    marginTop: spacing.sm,
  },
});

File: mobile-app-flutter/lib/screens/lock_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../auth/auth_provider.dart';
import '../theme/app_theme.dart';
import '../widgets/primary_button.dart';

class LockScreen extends StatefulWidget {
  const LockScreen({super.key});

  @override
  State<LockScreen> createState() => _LockScreenState();
}

class _LockScreenState extends State<LockScreen> {
  bool _loading = false;

  Future<void> _unlock() async {
    setState(() => _loading = true);
    try {
      final ok = await context.read<AuthProvider>().unlock();
      if (!ok && mounted) {
        ScaffoldMessenger.of(context).showSnackBar(
          const SnackBar(content: Text('Biometric authentication failed.')),
        );
      }
    } finally {
      if (mounted) setState(() => _loading = false);
    }
  }

  @override
  void initState() {
    super.initState();
    WidgetsBinding.instance.addPostFrameCallback((_) => _unlock());
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: AppColors.background,
      body: Center(
        child: Padding(
          padding: const EdgeInsets.all(40),
          child: Column(
            mainAxisSize: MainAxisSize.min,
            children: [
              const Icon(Icons.lock_outline,
                  size: 72, color: AppColors.primary),
              const SizedBox(height: 24),
              const Text(
                'SchoolDom is locked',
                style: TextStyle(
                    color: AppColors.text,
                    fontSize: 24,
                    fontWeight: FontWeight.w900),
              ),
              const SizedBox(height: 8),
              const Text(
                'Use biometrics or your device passcode to continue.',
                textAlign: TextAlign.center,
                style: TextStyle(color: AppColors.muted),
              ),
              const SizedBox(height: 40),
              PrimaryButton(
                title: 'Unlock',
                onPressed: _unlock,
                loading: _loading,
              ),
              const SizedBox(height: 12),
              TextButton(
                onPressed: () => context.read<AuthProvider>().signOut(),
                child: const Text('Sign out instead',
                    style: TextStyle(color: AppColors.muted)),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
